crowbar_framework/app/models/node.rb

```diff
@@ -302,6 +302,13 @@ def ready?
     state === "ready"
   end
 
+  # Right after node upgrade, chef-client is not yet running thus the state
+  # might not be updated for some time.
+  def ready_after_upgrade?
+    s = state
+    s == "ready" || s == "noupdate"
```

Comment:
I am pretty sure that doesn't happen when you use ```include?``` as the return value of the ```state``` method will be passed as parameter to the ```include?``` function.